x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -598,32 +598,40 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
   );
 
   const editRule = useMemo(() => {
-    if (!hasActionsPrivileges) {
-      return (
-        <EuiToolTip position="top" content={ruleI18n.EDIT_RULE_SETTINGS_TOOLTIP}>
-          <LinkButton
-            onClick={goToEditRule}
-            iconType="controlsHorizontal"
-            isDisabled={true}
-            href={formatUrl(getEditRuleUrl(ruleId ?? ''))}
-          >
-            {ruleI18n.EDIT_RULE_SETTINGS}
-          </LinkButton>
-        </EuiToolTip>
-      );
-    }
-    return (
+    const renderEditRuleSettingButtonLink = (disabled: boolean) => (
```

Comment:
Can we make it a React component? Something like `<EditRuleSettingButtonLink />` would feel more natural than a function that returns JSX.

Reply:
right, it's a good thought. I've done that in my new commit.

Reply:
Thanks, that looks much better now 🙌